### PR TITLE
Add api metadata yaml support for auto documentation tools

### PR DESCRIPTION
**Pull request by jamestyack**
The added YAML fields are optional and can be used to help document the Dropwizard Spring application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,16 @@ This is required to have maven build a "fat," executable jar file.
 
 	# dropwizard-spring sample service configuration
 
+	# High level information that describes the application. This can be pulled into auto documentation tools like Swagger.
+	apiMetadata:
+	    rootServerName: www.someshop.zzz
```
Comment (bflad) [inline README.md]: Please use standard www.example.com

File: README.md [hunk 2]
```diff
@@ -73,6 +73,16 @@ This is required to have maven build a "fat," executable jar file.
...
+	    title: My Online Shop
+	    description: Configure and use the online shop via this API
+	    contact: developers@someshop
```
Comment (bflad) [inline README.md]: @example.com

File: README.md [hunk 3]
```diff
@@ -73,6 +73,16 @@ This is required to have maven build a "fat," executable jar file.
...
+	    license: Licence type here
+	    licenseUrl: http://urltolicence.zzz
```
Comment (bflad) [inline README.md]: http://www.example.com/licence

File: README.md [hunk 4]
```diff
@@ -73,6 +73,16 @@ This is required to have maven build a "fat," executable jar file.
...
+	    termsOfServiceUrl: http://urltotos.zzz
```
Comment (bflad) [inline README.md]: http://www.example.com/tos

**Comment by bflad** [comment]
example.com fixes, otherwise LGTM

**Comment by boneill42** [comment]
Agreed.  LGTM aside from the notes from @bflad

**Comment by jamestyack** [comment]
Agree, updating to example.com

**Pull request merged**